inc/import/parser/class.ParserFITSingle.php

```diff
@@ -10,668 +10,134 @@
-				break;
-			case 'lap':
-				$this->readFieldDescriptionFor($this->nativeFieldMappingForLap, $this->DeveloperFieldMappingForLap);
-				break;
-		}
-	}
-
-	protected function readFieldDescriptionFor(array &$nativeFieldMapping, array &$fieldMapping) {
-		if (
-			isset($this->Values['native_field_num']) &&
-			isset($nativeFieldMapping[$this->Values['native_field_num'][0]]) &&
-			!empty($nativeFieldMapping[$this->Values['native_field_num'][0]][0]) &&
-			isset($this->Values['developer_data_index']) &&
-			isset($this->Values['field_name'])
-		) {
-			$fieldname = $this->Values['developer_data_index'][0].'_'.str_replace(['"', ' '], ['', '_'], $this->Values['field_name'][0]);
-			$fieldname = preg_replace_callback('/(\W)/i', function(array $char) {
-			    return sprintf('_%02x_', ord($char[0]));
-			}, preg_replace('/(\s+)/i', '_', $fieldname));
-			
-			$nativeFieldNum = $this->Values['native_field_num'][0];
-			$unitDefinition = str_replace('"', '', $this->Values['units'][0]);
-
-			$mappingKey = $nativeFieldMapping[$nativeFieldNum][0];
-			$mappingFactor = isset($nativeFieldMapping[$nativeFieldNum][1]) ? $nativeFieldMapping[$nativeFieldNum][1] : 1;
-
-			if (is_array($mappingFactor)) {
-				$mappingFactor = isset($mappingFactor[$unitDefinition]) ? $mappingFactor[$unitDefinition] : $mappingFactor['default'];
-			}
-
-			if (is_array($mappingKey)) {
-				$mappingKey = array_shift($nativeFieldMapping[$nativeFieldNum][0]);
-			}
-
-			$fieldMapping[$fieldname] = [$mappingKey, $mappingFactor];
-		}
+	public function setFitData(adriangibbons\phpFITFileAnalysis $fit) {
+		$this->fitData = $fit;
 	}
 
 	/**
 	 * Parse
+	 * @throws \Runalyze\Import\Exception\ParserException
 	 */
 	public function parse() {
-		// Uses another interface to not hold the complete file
-	}
-
-	/**
-	 * Start a new activity at current point
-	 */
-	public function startNewActivity() {
-		$creator = $this->TrainingObject->getCreator();
-		$creatorDetails = $this->TrainingObject->getCreatorDetails();
-		$offset = $this->TrainingObject->getTimezoneOffset();
+		$creatorDetails = '';
 
 		$this->TrainingObject = new TrainingObject(DataObject::$DEFAULT_ID);
 		$this->TrainingObject->setTimestamp(PHP_INT_MAX);
-		$this->TrainingObject->setTimezoneOffset($offset);
-		$this->TrainingObject->setCreator($creator);
-		$this->TrainingObject->setCreatorDetails($creatorDetails);
-
-		$this->isPaused = false;
-		$this->wasPaused = false;
-		$this->isSwimming = false;
-		$this->PauseInSeconds = 0;
-		$this->lastStopTimestamp = false;
-
-		foreach (array_keys($this->gps) as $key) {
-			$this->gps[$key] = array();
-		}
-	}
-
-	/**
-	 * Finish parsing
-	 */
-	public function finishParsing() {
-		$this->applyPauses();
-		$this->setGPSarrays();
-		$this->fixForSuunto();
-
-		// TODO: lookup timezone and correct timestamp if startpoint is not in user's timezone
-	}
+		$this->TrainingObject->setTimezoneOffset(0);
```

Comment:
Time zone offset should be `null` if it's unknown. Setting it to `0` assumes that it is known.